packit_service/service/events.py

```diff
@@ -699,6 +698,10 @@ def tag_name(self):
             self._tag_name = releases[0].tag_name if releases else ""
         return self._tag_name
 
+    @property
+    def commit_sha(self):
+        return self.tag_name
```

Comment:
I know that I am late to the party, but this should really be a `commit_sha`. (We have `identifier` for the event name.)

(It should work but can be misleading and we can have troubles later: e.g. tags needs to be pulled, unlike the commits.)

Reply:
```python
    @property
        if not self._tag_name:
            releases = self.project.get_releases()
            self._tag_name = releases[0].git_tag.commit_sha if releases else ""
        return self._tag_name
```

Or some optimisation to not need to search for releases for twice.

Reply:
Yeah, I am aware that this was a gross workaround but the way inheritance is set up in these events is really inconsistent and scary and don't have the bandwidth to fix it now.

If you look around the inheritance tree you'll see commit_sha, tag_name, git_ref, base_branch and all kind of weird stuff which probably could be unified. Unfortunately it's not clear based on what these attributes were assigned to a given layer of the inheritance tree, and it's also nowhere documented what their intended usage is.

The ideal solution would be to have a `git_ref` at the base and have it set to sha or branch or tag.

But on the long run I would like to see the `Event` classes to be gone (the way we do celery tasks now allows this), so from my point of view, there is no big hurry to fix this.

Reply:
I understand your confusion, but this makes it much worse since the name and value do not match. And the solution is really easy as you can see from my proposal.

There are three common attributes for all events, each used in a different way:
- `commit_sha` - commit hash
- `git_ref` - git checkout-able reference
- `identifier` - can be shown to the user

One of the reasons it's like that is because commit of the PR cannot be easily checkout-ed without additional work.

Your points make sense but are not relevant to this. My point is to match attribute naming, which is not so hard.

Reply:
Filed #744 to fix this.

Reply:
Thanks